sdk/cognitivelanguage/azure-ai-language-questionanswering/migration_guide.md

```diff
@@ -60,70 +69,381 @@ client = QuestionAnsweringClient(
+```
+
+#### Updating knowledge base
+
+Previously in `azure-cognitiveservices-knowledge-qnamaker`, you could update your knowledge base using a `UpdateKbOperationDTO`:
+
+```python
+from msrest.authentication import CognitiveServicesCredentials
+from azure.cognitiveservices.knowledge.qnamaker.models import UpdateKbOperationDTO, UpdateKbOperationDTOAdd, QnADTO
+from azure.cognitiveservices.knowledge.qnamaker import QnAMakerClient
+
+client = QnAMakerClient(
+    endpoint="https://<my-cognitiveservices-account>.cognitiveservices.azure.com",
+    credentials=CognitiveServicesCredentials("API key")
+)
+
+operation = client.knowledgebase.update(
+    kb_id="<knowledgebase-id>",
+    update_kb=UpdateKbOperationDTO(
+        add=UpdateKbOperationDTOAdd(
+            qna_list=[
+                QnADTO(
+                    questions=["questions"],
+                    answer="answer"
+                )
+            ]
+        )   
+    )
+)
+```
+
+Now in `azure-ai-language-questionanswering`, you can update your knowledge source using the `begin_update_sources` method:
+
+```python
+from azure.core.credentials import AzureKeyCredential
+from azure.ai.language.questionanswering.authoring import QuestionAnsweringAuthoringClient
+
+client = QuestionAnsweringAuthoringClient(
+    endpoint="https://<my-cognitiveservices-account>.cognitiveservices.azure.com",
+    credential=AzureKeyCredential("API key")
+)
+
+sources_poller = client.begin_update_sources(
+    project_name="<project_name>",
+    sources=[{
+        "op": "add",
+        "value": {
+            "displayName": "MicrosoftFAQ",
+            "source": "https://www.microsoft.com/en-in/software-download/faq",
+            "sourceUri": "https://www.microsoft.com/en-in/software-download/faq",
+            "sourceKind": "url",
+            "contentStructureKind": "unstructured",
+            "refresh": False
+        }
+    }]
+)
+sources = sources_poller.result()
+
+for item in sources:
+    print("source name: {}".format(item.get("displayName", "N/A")))
+    print("\tsource: {}".format(item["source"]))
+    print("\tsource uri: {}".format(item.get("sourceUri", "N/A")))
+    print("\tsource kind: {}".format(item["sourceKind"]))
+```
+
+You can also update a project's questions and answers directly as follows:
+
+```python
+from azure.core.credentials import AzureKeyCredential
+from azure.ai.language.questionanswering.authoring import QuestionAnsweringAuthoringClient
+
+client = QuestionAnsweringAuthoringClient(
+    endpoint="https://<my-cognitiveservices-account>.cognitiveservices.azure.com",
+    credential=AzureKeyCredential("API key")
+)
+
+qna_poller = client.begin_update_qnas(
+    project_name="<project_name>",
+    qnas=[{
+        "op": "add",
+        "value": {
+            "questions": [
+                "What is the easiest way to use Azure services in my Python project?"
+            ],
+            "answer": "Using the Azure SDKs"
+        }
+    }]
+)
+
+qnas = qna_poller.result()
```

Comment:
Conversely, here you get results from an operation/poller. Others should be similar. Should the poller be named `operation`, though, or the others a `poller`? Just for consistency, which should help with migration.